File: backend/controllers/order_controller.py
from flask import Blueprint, request, jsonify
from models import OrderModel

order_bp = Blueprint('order', __name__)

@order_bp.route('/api/checkout', methods=['POST'])
def checkout():
    data = request.json
    cart = data.get('cart', [])
    username = data.get('username', 'Guest')

    order, error = OrderModel.create_order(username, cart)
    if order:
        return jsonify({'success': True, 'order': order})
    return jsonify({'success': False, 'message': error}), 400

@order_bp.route('/api/orders', methods=['GET'])
def get_orders():
    orders = OrderModel.get_all_orders()
    return jsonify(orders)

@order_bp.route('/api/customer/orders', methods=['POST'])
def get_customer_orders():
    data = request.json
    username = data.get('username')
    if not username:
        return jsonify({'success': False, 'message': 'Username required'}), 400
    orders = OrderModel.get_orders_by_username(username)
    return jsonify({'success': True, 'orders': orders})

File: backend/src/controllers/orderController.js
const orderService = require("../services/orderService");

async function checkout(req, res) {
  const { order, error } = await orderService.createOrder({
    cart: Array.isArray(req.body.cart) ? req.body.cart : [],
    username: req.body.username || "Guest"
  });

  if (error) {
    res.status(400).json({ success: false, message: error });
    return;
  }

  res.json({ success: true, order });
}

async function getOrders(req, res) {
  const orders = await orderService.listOrders();
  res.json(orders);
}

async function getCustomerOrders(req, res) {
  if (!req.body.username) {
    res.status(400).json({ success: false, message: "Username required" });
    return;
  }

  const orders = await orderService.listOrdersByUsername(req.body.username);
  res.json({ success: true, orders });
}

module.exports = {
  checkout,
  getCustomerOrders,
  getOrders
};

